wcivf/apps/elections/tests/test_election_views.py

```diff
@@ -141,6 +141,117 @@ def test_num_candidates(self):
             response, f"See all 5 candidates in the {self.post.label}"
         )
 
+    def test_cancelled_election_with_candidates(self):
+        self.post_election.winner_count = 4
+        people = [PersonFactory() for p in range(4)]
+        for person in people:
+            PersonPostFactory(
+                post_election=self.post_election,
+                election=self.election,
+                post=self.post,
+                person=person,
+            )
+        self.post_election.cancelled = True
+        self.post_election.save()
+        response = self.client.get(
+            self.post_election.get_absolute_url(), follow=True
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "elections/post_view.html")
+        self.assertTemplateUsed(
+            response, "elections/includes/_post_meta_title.html"
+        )
+        self.assertTemplateUsed(
+            response, "elections/includes/_post_meta_description.html"
+        )
+        self.assertContains(response, "Cancelled Election")
+        self.assertNotContains(response, "No candidates known yet.")
+
+    def test_cancelled_election_without_candidates(self):
+        self.post_election.winner_count = 4
+        self.post_election.cancelled = True
+        self.post_election.save()
+        response = self.client.get(
+            self.post_election.get_absolute_url(), follow=True
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "elections/post_view.html")
+        self.assertTemplateUsed(
+            response, "elections/includes/_post_meta_title.html"
+        )
+        self.assertTemplateUsed(
+            response, "elections/includes/_post_meta_description.html"
+        )
+        self.assertContains(response, "Cancelled Election")
+        self.assertNotContains(response, "No candidates known yet.")
+
+    # In the case of fewer candidates than seats but where number of candidates is one or more
+    def test_uncontested_and_rescheduled(self):
+        self.post_election.winner_count = 5
+        people = [PersonFactory() for p in range(4)]
+        for person in people:
+            PersonPostFactory(
+                post_election=self.post_election,
+                election=self.election,
+                post=self.post,
+                person=person,
+            )
+        self.post_election.uncontested = True
```

Comment:
There is no `uncontested` field on `PostElection` think this should be `self.post_election.contested = False`

Reply:
https://github.com/DemocracyClub/WhoCanIVoteFor/blob/master/wcivf/apps/elections/models.py#L342